networkx/generators/geometric.py

```diff
@@ -281,6 +281,11 @@ def soft_random_geometric_graph(
     exponential distribution with rate parameter :math:`\lambda=1` if their
     Euclidean distance is at most 0.2.
 
+    >>> from scipy.stats import expon
+    >>> n = 100
+    >>> p_dist = lambda x: expon.pdf(x, scale=1)
+    >>> G = nx.soft_random_geometric_graph(n, 0.2, p_dist=p_dist)
+
```

Comment:
Thanks for adding this @yuanx749 ! If possible, please consider moving the example(explaining the usage of `pos` kwarg) from the `Notes` section to this `Example` section. And the whole `Example` section at the end(https://numpydoc.readthedocs.io/en/latest/format.html#examples). Thanks!

Reply:
Thanks for your review and addressing the CI failure. @Schefflera-Arboricola . I've made the move, with wording slightly adjusted.